libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/filescan/FileSystemScanExecutor.java

```diff
@@ -136,6 +142,111 @@ public void execute() {
+                                LOG.error("Artifact " + foundArtifact.getId() + " does not contain asset " + assetId);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // 3. build zips for all components
+        buildZipsForAllComponents(baseDir, filePatternQualifierMappers);
+    }
+
+    public void buildZipsForAllComponents(File baseDir, List<FilePatternQualifierMapper> filePatternQualifierMappers) {
+        // Create an Ant project
+        Project antProject = new Project();
+        antProject.init();
+
+        for (FilePatternQualifierMapper mapper : filePatternQualifierMappers) {
+            // filter out big tar files
+            mapper.getFileMap().values().forEach(files -> files.removeIf(f -> f.length() > 100000000));
+
+            // Loop over each entry in the file map
+            for (Map.Entry<Boolean, List<File>> entry : mapper.getFileMap().entrySet()) {
+                List<File> files = entry.getValue();
+                if (files.isEmpty()) {
+                    continue;
+                }
+
+                final File targetDir = new File(baseDir, "components");
+                if (!targetDir.exists()) {
+                    targetDir.mkdirs();
+                }
+
+                final File tmpFolder = initializeTmpFolder(targetDir);
+
+                final File zipFile = new File(targetDir, mapper.getQualifier() + ".zip");
+                File relativeBaseDir;
+                if (mapper.getPathInAsset().contains(".|\n")) {
+                    relativeBaseDir = new File(baseDir.getPath(), ".");
+                } else {
+                    relativeBaseDir = new File(baseDir.getPath(), mapper.getPathInAsset());
+                }
+
+                // Create a new Ant Zip task
+                Zip zipTask = new Zip();
+                zipTask.setProject(antProject);
+                zipTask.setDestFile(zipFile);
+
+                // Add each file to the zip
+                for (File file : files) {
+                    ZipFileSet zipFileSet = new ZipFileSet();
+
+                    if (file.isFile()) {
+                        zipFileSet.setFile(file);
+                    } else if (file.isDirectory()) {
+                        zipFileSet.setDir(file);
+                    }
+
+                    try {
+                        FileUtils.copyFile(file, new File(tmpFolder, file.getName()));
+                    } catch (IOException e) {
+                        LOG.error("Failed to copy file to tmp folder.", e);
+                    }
+
+                    // Set the prefix to preserve the relative directory structure
+                    String relativePath = relativeBaseDir.toURI().relativize(file.getParentFile().toURI()).getPath();
+                    zipFileSet.setPrefix(relativePath);
+
+                    zipTask.addZipfileset(zipFileSet);
```

Comment:
Why don't you copy the folder and then zip it as a whole. code for this already exists in parts.

Reply:
wouldn't I have problems with contained artifacts? so, lets say if subset B is in set A and I copy the folder of A and zip it, i would duplicate those files. since I don't physically remove those files from the filesystem and instead collect it in a separate file list and remove it from this list.